src/main/java/uk/gov/hmcts/reform/em/stitching/pdf/PDFMerger.java

```diff
@@ -31,13 +39,25 @@ public File merge(Bundle bundle, Map<BundleDocument, File> documents) throws IOE
         private final Bundle bundle;
         private static final String BACK_TO_TOP = "Back to top";
         private int currentPageNumber = 0;
+        private TemplateRenditionClient templateRenditionClient;
 
-        public StatefulPDFMerger(Map<BundleDocument, File> documents, Bundle bundle) {
+        public StatefulPDFMerger(Map<BundleDocument, File> documents, Bundle bundle,
+                                 TemplateRenditionClient templateRenditionClient) {
             this.documents = documents;
             this.bundle = bundle;
+            this.templateRenditionClient = templateRenditionClient;
         }
 
-        public File merge() throws IOException {
+        public File merge() throws IOException, DocumentTaskProcessingException {
+            if (StringUtils.isNotBlank(bundle.getCoverpageTemplate())) {
```

Comment:
I would prefer if the rendition happened outside of the PdfMerger and a rendered PDF file was passed in. Just to reduce the amount of logic and work that happens in here.

Reply:
This has been refactored 👍